src/components/text/marked/index.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { FC } from 'react';
+import { Box, BoxProps, useMultiStyleConfig } from '@chakra-ui/react';
+
+import Highlight from './highlight';
+
+type SharedProps = Exclude<
+  BoxProps,
+  'position' | 'width' | 'display' | 'alignItems'
+>;
+
+type HighlightProps = SharedProps & {
+  variant?: 'highlight';
+  highlightColor?: 'white' | 'brand';
+};
+
+type UnderlineProps = SharedProps & {
+  variant: 'underline';
+  highlightColor?: never;
+};
+
+export type MarkedTextProps = HighlightProps | UnderlineProps;
+
+const MarkedText: FC<MarkedTextProps> = ({
+  children,
+  variant = 'highlight',
+  highlightColor,
+  ...boxProps
+}) => {
+  const { container, text } = useMultiStyleConfig('MarkedText', {
+    variant,
+    highlightColor,
+  });
+
+  return (
+    <Box __css={container} {...boxProps}>
+      <Highlight variant={variant} highlightColor={highlightColor} />
+      <Box __css={text}>{children}</Box>
```

Comment:
Should we pass children here or a text prop? Thinking it might get wired results if we pass something other than plain text?

Reply:
I opted for consistency with `Text` component that takes the children. We could technically type children as `string` (or use a prop), but this approach gives us a bit more flexibility regarding formatting. You could to something like this:

```jsx

<MarkedText>
  <Box as="span" fontWeight="bold" color="blue" px="60px">
     bold blue with some padding
  </Box>
</MarkedText>
```
which would be cumbersome otherwise